pilot/pkg/config/kube/gateway/conversion.go

```diff
@@ -798,8 +798,7 @@ func toInternalParentReference(p k8s.ParentReference, localNamespace string) (pa
 	// Currently supported types are Gateway and Service
 	if kind == gvk.KubernetesGateway.Kind && nilOrEqual((*string)(p.Group), gvk.KubernetesGateway.Group) {
 		ik = gvk.KubernetesGateway
-	} else if kind == gvk.Service.Kind && (nilOrEqual((*string)(p.Group), gvk.Service.Group) ||
-		*(*string)(p.Group) == gvk.KubernetesGateway.Group) { // TODO: gateway group is default?
+	} else if kind == gvk.Service.Kind && *(*string)(p.Group) == gvk.Service.Group {
```

Comment:
donot it break users that donot set group previously

Reply:
and panic

Reply:
I realize this is a breaking change, but the gateway-api for mesh is still alpha/experimental, so we should change it now IMO, before we have a backward compatibility requirement. The gateway-api spec is very clear now that the explicit `group: ""` is required:
```
Group is the group of the referent. When unspecified, “gateway.networking.k8s.io” is inferred.
To set the core API group (such as for a “Service” kind referent), Group must be explicitly set
to “” (empty string).
```